ek9000App/src/devEK9000.cpp

```diff
@@ -1457,167 +1494,231 @@ int CoE_ParseString(const char* str, ek9k_coe_param_t* param) {
 	else if (epicsStrnCaseCmp(buffers[4], "int8", 4) == 0 || epicsStrnCaseCmp(buffers[4], "uint8", 5) == 0)
 		param->type = ek9k_coe_param_t::COE_TYPE_INT8;
 	else
-		return 1;
+		return false;
 
 	termid = atoi(buffers[1]);
 	if (errno != 0)
-		return 1;
+		return false;
 	pterm = pcoupler->m_terms[termid - 1];
 
 	param->pterm = pterm;
 	param->ek9k = pcoupler;
 	param->index = (int)strtol(buffers[2], 0, 16);
 	if (errno != 0)
-		return 1;
+		return false;
 
 	param->subindex = (int)strtol(buffers[3], 0, 16);
 	if (errno != 0)
-		return 1;
+		return false;
 
-	return 0;
+	return true;
 }
 
 //-----------------------------------------------------------------//
 // EK9K RO configuration/status
-static long ek9k_ro_init(int pass);
-static long ek9k_ro_init_record(void* prec);
-static long ek9k_ro_read_record(void* pprec);
+static long ek9k_status_init(int pass);
+template <class T> static long ek9k_status_init_record(void* prec); // Common init function for status/config records
+static long ek9k_status_read_record(void* prec);
+static long ek9k_status_write_record(void* prec);
+static long ek9k_status_get_ioint_info(int cmd, void* prec, IOSCANPVT* iopvt);
+
+static bool ek9k_parse_string(const char* linkValue, ek9k_param_t& outParam);
+
+enum {
+	STATUS_RD = 0x1,
+	STATUS_WR = 0x2,
+	STATUS_RW = STATUS_RD | STATUS_WR,
+	STATUS_STATIC = 0x4, /* These registers will never change during runtime, only need to read these once */
+};
+typedef int StatusFlags;
 
+struct StatusReg {
+	const char* configName;
+	int addr;
+	StatusFlags flags;
+};
+
+CONSTEXPR StatusReg status_regs[] = {{"analogOutputs", 0x1010, STATUS_RD | STATUS_STATIC},
+									 {"analogInputs", 0x1011, STATUS_RD | STATUS_STATIC},
```

Comment:
Having these by name is convenient for sure.  👍 

Are there any reserved values in this status register block that the user may end up wanting to access down the line? If you think there's potential for it, it may be beneficial to offer the ability to specify the status register address/flags in the instio string itself so they don't have to rewrite the EK9K code to allow for it.

Reply:
Probably a good idea. This does cover virtually all of the status registers though, except 0x6000-0x60FF, which contain rail layout.